golos.referral/include/golos.referral/golos.referral.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+#include <eosiolib/eosio.hpp>
+#include <eosiolib/asset.hpp>
+#include <eosiolib/time.hpp>
+
+
+namespace golos {
+
+using namespace eosio;
+
+class referral: public contract {
+public:
+    using contract::contract;
+
+    [[eosio::action]]
+    void addreferral(name referrer, name referral, uint32_t percent, uint64_t expire, asset breakout);
+
+private:
+    struct obj_referral {
+        uint64_t id;
```

Comment:
`id` field is not required for this table. You can use `referral` field as primary-key. Each referral can have only one referrer and so this field are unique.